src/upgrade.c

```diff
@@ -322,16 +322,16 @@ exec_upgrade(int argc, char **argv)
 
 	if (dry_run && !auto_update)
 		retcode = pkgdb_access(PKGDB_MODE_READ,
-				       PKGDB_DB_LOCAL|PKGDB_DB_REPO);
+				       PKGDB_DB_LOCAL|PKGDB_DB_REPO, reponame, NULL);
 	else
 		retcode = pkgdb_access(PKGDB_MODE_READ  |
 				       PKGDB_MODE_WRITE |
 				       PKGDB_MODE_CREATE,
-				       PKGDB_DB_LOCAL|PKGDB_DB_REPO);
+				       PKGDB_DB_LOCAL|PKGDB_DB_REPO, reponame, NULL);
 	if (retcode == EPKG_ENOACCESS && dry_run) {
 		auto_update = false;
 		retcode = pkgdb_access(PKGDB_MODE_READ,
-				       PKGDB_DB_LOCAL|PKGDB_DB_REPO);
+				       PKGDB_DB_LOCAL|PKGDB_DB_REPO, reponame, NULL);
```

Comment:
This will turn into `(PKGDB_DB_LOCAL|PKGDB_DB_REPO, "", NULL);` for the simple `pkg upgrade` command. I take it, this is not the same as `PKGDB_DB_LOCAL|PKGDB_DB_REPO, NULL);`?

Reply:
Isn't `reponame` NULL by default?

Reply:
Of course, `(PKGDB_DB_LOCAL|PKGDB_DB_REPO, "", NULL)` is not the same as `(PKGDB_DB_LOCAL|PKGDB_DB_REPO, NULL, NULL)`, but from what I see `reponame` is NULL by default so it should all be correct.

Reply:
yes reponame is null by default